src/proto/message_translation/power_frame_msg.hpp

```diff
@@ -0,0 +1,218 @@
+}
+#endif
+
+// The nanopb generated size isn't c++ compatible so we redefine it here
+// TODO(#2592): Remove with upgrade to nanopb
+#undef TbotsProto_PowerFrame_size
+#define TbotsProto_PowerFrame_size                                                       \
+    std::max(TbotsProto_PowerControl_size, TbotsProto_PowerStatus_size) +                \
+        2 * sizeof(uint32_t) + sizeof(uint16_t)
+
+
+/**
+ * Serialize nanopb into its byte representation
+ *
+ * @param data nanopb msg to be serialized
+ * @return vector of bytes representation of provided nanopb
+ */
+template <typename T>
+std::vector<uint8_t> serializeToVector(const T& data)
+{
+    const pb_field_t* fields;
+    int size;
+    if (std::is_same<T, TbotsProto_PowerFrame>::value)
+    {
+        fields = TbotsProto_PowerFrame_fields;
+        size   = TbotsProto_PowerFrame_size;
+    }
+    else if (std::is_same<T, TbotsProto_PowerControl>::value)
+    {
+        fields = TbotsProto_PowerControl_fields;
+        size   = TbotsProto_PowerControl_size;
+    }
+    else if (std::is_same<T, TbotsProto_PowerStatus>::value)
+    {
+        fields = TbotsProto_PowerStatus_fields;
+        size   = TbotsProto_PowerStatus_size;
+    }
+    else
+    {
+        throw std::runtime_error("Unexpected type as argument");
+    }
+
+    std::vector<uint8_t> buffer(size);
+    pb_ostream_t stream =
+        pb_ostream_from_buffer(static_cast<uint8_t*>(buffer.data()), buffer.size());
+    if (!pb_encode(&stream, fields, &data))
+    {
+        throw std::runtime_error(
+            "Failed to encode PowerFrame msg to vector when converting nanopb to vector");
+    }
+    return buffer;
+}
+
+/**
+ * Helper function to set the type and value of a power frame's power_msg
+ *
+ * @param frame frame to set the power_msg of
+ * @param control/status power_msg to set
+ */
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerControl& control)
+{
+    frame.which_power_msg         = TbotsProto_PowerFrame_power_control_tag;
+    frame.power_msg.power_control = control;
+}
+
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerStatus& status)
+{
+    frame.which_power_msg        = TbotsProto_PowerFrame_power_status_tag;
+    frame.power_msg.power_status = status;
+}
+
+/**
+ * Creates a nanopb power status msg with provided fields
+ *
+ * @return a nanopb power status msg with provided fields
+ */
+TbotsProto_PowerStatus createNanoPbPowerStatus(float battery_voltage,
+                                               float capacitor_voltage,
+                                               float current_draw,
+                                               float high_voltage_measurement_volts,
+                                               float geneva_angle_deg,
+                                               bool breakbeam_tripped, bool flyback_fault)
+{
+    TbotsProto_PowerStatus status         = TbotsProto_PowerStatus_init_default;
```

Comment:
we can probably initialize like so and avoid the constructor:
TbotsProto_PowerStatus status{.battery_voltage = 1, ....}
See https://en.cppreference.com/w/c/language/struct_initialization